consensus/hotstuff/votecollector/combined_vote_processor_v3.go

```diff
@@ -242,10 +267,18 @@ func (p *CombinedVoteProcessorV3) Process(vote *model.Vote) error {
 // signatures for building a QC. This function is run only once by a single worker.
 // Any error should be treated as exception.
 func (p *CombinedVoteProcessorV3) buildQC() (*flow.QuorumCertificate, error) {
+	// STEP 1: aggregate staking signatures (if there are any)
+	// Note: it is possible that all replicas signed with their random beacon keys. In this case, the Aggregator
+	// returns an `model.InsufficientSignaturesError`. If no replicas signed with their staking key,
+	// `BlockSignatureData.StakingSigners` and `BlockSignatureData.AggregatedStakingSig` should be set to nil.
 	stakingSigners, aggregatedStakingSig, err := p.stakingSigAggtor.Aggregate()
-	if err != nil {
-		return nil, fmt.Errorf("could not aggregate staking signature: %w", err)
+	if err != nil && !model.IsInsufficientSignaturesError(err) {
```

Comment:
Context:
* This is consensus voting V3, i.e. consensus participants vote with _either_ their staking signature _or_ their random beacon signature. 
* Currently, all consensus nodes are also random beacon members. Ideally, all consensus nodes would successfully participate in the DKG and then only sign their vote with their random beacon keys. 

From the perspective of the `CombinedVoteProcessorV3`:
* all votes are processed by the [`rbSigAggtor` and  `rbRector`](https://github.com/onflow/flow-go/blob/c6fd13ec8bf7adf51f631ae12a2cfcef7780a840/consensus/hotstuff/votecollector/combined_vote_processor_v3.go#L202-L227)
* There is not a single signature added to the `stakingSigAggtor`. 
* While the  `stakingSigAggtor`'s total weight remains `0`, the `rbSigAggtor` can easily reach the required weight for consensus supermajority. (-> [code](https://github.com/onflow/flow-go/blob/c6fd13ec8bf7adf51f631ae12a2cfcef7780a840/consensus/hotstuff/votecollector/combined_vote_processor_v3.go#L234))
* As all the voters sign with their random beacon keys, the required threshold for reconstructing the group signature is also easily reached (->[code](https://github.com/onflow/flow-go/blob/c6fd13ec8bf7adf51f631ae12a2cfcef7780a840/consensus/hotstuff/votecollector/combined_vote_processor_v3.go#L237-L239))

At this point, we have collected enough votes to construct a valid QC. We proceed to call `buildQC`: https://github.com/onflow/flow-go/blob/c6fd13ec8bf7adf51f631ae12a2cfcef7780a840/consensus/hotstuff/votecollector/combined_vote_processor_v3.go#L269-L274
where `stakingSigAggtor.Aggregate()` returns with an `InsufficientSignaturesError`
* The `stakingSigAggtor` is essentially telling us, that it can't build an aggregated signature, because it doesn't have any signatures at all.
* At the point where we want to build the QC, we have to decide how we represent the empty set of staking signers.

Reply:
I have updated the logic for better readability (commit https://github.com/onflow/flow-go/pull/1830/commits/40321dfb4df551354015f098e18490c5a094b811 ) as follows: https://github.com/onflow/flow-go/blob/40321dfb4df551354015f098e18490c5a094b811/consensus/hotstuff/votecollector/combined_vote_processor_v3.go#L269-L287

I have this is clearer and better documented.

Reply:
updated the property test to also cover this case (-> [revised code](https://github.com/onflow/flow-go/blob/adb2de0ec9114e2d44f8fb8b5ceb04ee34e37080/consensus/hotstuff/votecollector/combined_vote_processor_v3_test.go#L519-L557))

Reply:
Totally makes sense, thanks for explanation.